src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/AdminTranscriptionServiceImpl.java

```diff
@@ -134,4 +152,56 @@ public GetTranscriptionDocumentByIdResponse getTranscriptionDocumentById(Integer
             throw new DartsApiException(TranscriptionApiError.TRANSCRIPTION_DOCUMENT_ID_NOT_FOUND);
         }
     }
+
+    @Transactional
+    public TranscriptionDocumentHideResponse hideOrUnhideTranscriptionDocumentById(Integer transcriptionDocumentId,
+                                                                         TranscriptionDocumentHideRequest transcriptionDocumentHideRequest) {
+
+        TranscriptionDocumentHideResponse response;
+
+        IdRequest<TranscriptionDocumentHideRequest> request = new IdRequest<>(transcriptionDocumentHideRequest, transcriptionDocumentId);
+        transcriptionDocumentHideOrShowValidator.validate(request);
+
+        Optional<TranscriptionDocumentEntity> transcriptionDocumentEntity
+            = transcriptionDocumentRepository.findById(transcriptionDocumentId);
+        TranscriptionDocumentEntity documentEntity = transcriptionDocumentEntity.get();
+
+        documentEntity.setHidden(transcriptionDocumentHideRequest.getIsHidden());
+        documentEntity = transcriptionDocumentRepository.saveAndFlush(documentEntity);
+
+        ObjectHiddenReasonEntity objectHiddenReasonEntity;
+        if (request.getPayload().getIsHidden()) {
+            objectHiddenReasonEntity = objectHiddenReasonRepository.findById(transcriptionDocumentHideRequest.getAdminAction().getReasonId()).get();
+
+            // on hiding add the relevant hide record
+            ObjectAdminActionEntity objectAdminActionEntity = new ObjectAdminActionEntity();
+            objectAdminActionEntity.setObjectHiddenReason(objectHiddenReasonEntity);
+            objectAdminActionEntity.setTicketReference(transcriptionDocumentHideRequest.getAdminAction().getTicketReference());
+            objectAdminActionEntity.setComments(transcriptionDocumentHideRequest.getAdminAction().getComments());
+            objectAdminActionEntity.setTranscriptionDocument(documentEntity);
+            objectAdminActionEntity.setHiddenBy(userIdentity.getUserAccount());
+            objectAdminActionEntity.setHiddenDateTime(OffsetDateTime.now());
+            objectAdminActionEntity.setMarkedForManualDeletion(false);
+            objectAdminActionEntity.setMarkedForManualDelBy(userIdentity.getUserAccount());
+            objectAdminActionEntity.setMarkedForManualDelDateTime(OffsetDateTime.now());
+
+            objectAdminActionEntity = objectAdminActionRepository.saveAndFlush(objectAdminActionEntity);
+
+            response = transcriptionMapper.mapHideOrShowResponse(documentEntity, objectAdminActionEntity);
+        } else {
+            List<ObjectAdminActionEntity> objectAdminActionEntityLst = objectAdminActionRepository.findByTranscriptionDocument_Id(transcriptionDocumentId);
```

Comment:
@chrisbellingham-hmcts in the case of a show i.e. not a hide should we return the existing admin action data (or is this only for hiding).

Reply:
Hey @mestebanez, so I think the scenario you're describing is "AC2 - Happy Path, Unhide a transcription document".

If we're going from Hidden to Unhidden, the response should reflect the new state of the system which would be Unhidden, i.e `is_hidden: false` with `admin_reason: null` (equivalent to no `admin_reason` being shown in the response).